src/utilities/new-sequence-editor/time-utils.ts

```diff
@@ -0,0 +1,250 @@
+export const ABSOLUTE_TIME = RegExp(/^(\d{4})-(\d{3})T(\d{2}):(\d{2}):(\d{2})(?:\.(\d{3}))?$/g);
+
+export const RELATIVE_TIME = /([0-9]{3}T)?([0-9]{2}):([0-9]{2}):([0-9]{2})(\.[0-9]+)?$/g;
+export const RELATIVE_SIMPLE = /(\d+)(\.[0-9]+)?$/g;
+
+export const EPOCH_TIME = /(^[+-]?)([0-9]{3}T)?([0-9]{2}):([0-9]{2}):([0-9]{2})(\.[0-9]+)?$/g;
+export const EPOCH_SIMPLE = /(^[+-]?)(\d+)(\.[0-9]+)?$/g;
+
+/**
+ * Tests if a given time string matches a specified regular expression.
+ *
+ * @param {string} time - The time string to be tested.
+ * @param {RegExp} regex - The regular expression to test against.
+ * @return {RegExpExecArray | null} The result of the regular expression execution, or null if no match is found.
+ */
+export function testTime(time: string, regex: RegExp): RegExpExecArray | null {
+  regex.lastIndex = 0;
```

Comment:
Is it necessary to reset the index like this? I think it can deleted.

Reply:
I'll defer to @goetzrrGit on this.

Reply:
Talked with @duranb using 'y' in the regex which preserves the index. He will make a ticket to address this later.